distributed/stealing.py

```diff
@@ -113,27 +114,21 @@ def steal_time_ratio(self, ts):
         For example a result of zero implies a task without dependencies.
         level: The location within a stealable list to place this value
         """
-        if not ts.dependencies:  # no dependencies fast path
-            return 0, 0
```

Comment:
Why not keep this?  My understanding is that this line says "if you don't have any dependencies then we can move you any time". This seems great to me :)

Reply:
That's very likely still happening. The only difference is that this now accounts for latencies.

Reply:
I figured: if the outcome is the same regardless, we would we need two if-clauses? (this one and the one further below). We do not really gain in speed either since it's all cheap, constant time operations below

Reply:
> We do not really gain in speed either since it's all cheap, constant time operations below

I think that it's reasonable to avoid these constant time operations.  Our budget per-task is currently around 200us over the lifetime of the task (and we're trying to reduce that).  There are enough attribute access and dict access calls that it seems non-trivial to me.

A fast path for the common case of no dependencies seems sensible here to me, although I haven't done any measurement to back that up.